### PR TITLE
systemd socket activation support

### PR DESCRIPTION
**Pull request by yawkat**
This PR adds support for binding to a systemd socket. It's a fairly minor internal change, but it could be useful.

### CODE REVIEW

File: src/main/docs/guide/httpServer/serverConfiguration/listener.adoc [hunk 1]
```diff
@@ -43,3 +43,48 @@ micronaut:
...
+  netty:
+    event-loops:
+      # use epoll
+      parent:
+        prefer-native-transport: true
+      default:
+        prefer-native-transport: true
+    listeners:
+      systemd:
+        fd: 3 # systemd passes the server socket as fd 3
```
Comment (timyates) [inline src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: Is this always `3`?

Or is this dependant on when systemd sets up the socket via the below config?

Are you on a Mac?  If so, how did you test this? ;-)

Comment (yawkat) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: it's always 3. and no, i'm on linux.

Comment (yawkat) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: https://github.com/systemd/systemd/blob/main/src/systemd/sd-daemon.h#L56

Comment (graemerocher) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: can we not default the value to 3 if it is always 3?

Comment (yawkat) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: i think it could be 0 if you were using inetd for example. the 3 is systemd-specific, but the fd config option is more generic.

Comment (yawkat) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: ive thought about detecting it from the env variables that systemd sets. but it would be a bit more complicated of a PR, there's no way to test it, it might have negative perf impact, and it is pretty niche in the first place.

this pr is basically the minimum to make it work with systemd with config options only, with no adverse impact since it's a niche feature.

Comment (timyates) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: > https://github.com/systemd/systemd/blob/main/src/systemd/sd-daemon.h#L56

If I'm reading that right, it starts at 3, but if you have multiple services, they'll be on `3 + n`?

Comment (yawkat) [reply src/main/docs/guide/httpServer/serverConfiguration/listener.adoc:63]: right. i dont really see a use case for micronaut there though, so i didnt add it to the docs.

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java [hunk 2]
```diff
@@ -1445,6 +1447,47 @@ public void setExposeDefaultRoutes(boolean exposeDefaultRoutes) {
...
+         * Whether the server should bind to the socket. {@code true} by default. If set to
+         * {@code false}, the socket must already be bound and listening.
+         *
+         * @return Whether to bind to the socket
+         */
+        public boolean isBind() {
+            return bind;
+        }
+
+
```
Comment (timyates) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java]:
```suggestion

```

**Review by timyates — APPROVED** [review]
👍 

I can't test it though, as I don't have Epoll :-(

**Review by xism4 — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]
